### PR TITLE
filter: add stats to aws lambda filter

### PR DESCRIPTION
**Pull request by marcomagdy**
Signed-off-by: Marco Magdy <mmagdy@gmail.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by lavignes** [comment]
Shouldn't we add these metrics to the lambda filter docs?

**Comment by marcomagdy** [comment]
Yes. I'll update it.

File: docs/root/configuration/http/http_filters/aws_lambda_filter.rst [hunk 1]
```diff
@@ -182,3 +182,18 @@ An example with the Lambda metadata applied to a weighted-cluster:
         "@type": type.googleapis.com/envoy.api.v2.auth.UpstreamTlsContext
         sni: "*.amazonaws.com"
 
+
+Statistics
```
Comment (lavignes) [inline docs/root/configuration/http/http_filters/aws_lambda_filter.rst:186]: thank you sir

**Review by lavignes — APPROVED** [review]
lgtm

**Comment by abaptiste** [comment]
lgtm

File: source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc [hunk 2]
```diff
@@ -416,6 +408,12 @@ absl::optional<Arn> parseArn(absl::string_view arn) {
   return Arn{partition, service, region, account_id, resource_type, function_name};
 }
 
+FilterStats generateStats(const std::string& prefix, Stats::Scope& scope) {
+  const std::string final_prefix = prefix + "aws_lambda.";
```
Comment (nealp9084) [inline source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc:412]: Correct me if I'm wrong, but this contradicts the documentation.

**Review by nealp9084 — APPROVED** [review]

**Comment by marcomagdy** [comment]
Thanks guys

**Pull request closed without merging**